cvmfs/swissknife_filestats.cc

```diff
@@ -0,0 +1,317 @@
+/**
+ * This file is part of the CernVM File System
+ */
+
+#include "swissknife_filestats.h"
+
+#include <cassert>
+
+#include "hash.h"
+#include "logging.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+using namespace std;  // NOLINT
+
+namespace swissknife {
+
+ParameterList CommandFileStats::GetParams() const {
+  ParameterList r;
+  r.push_back(Parameter::Mandatory(
+              'r', "repository URL (absolute local path or remote URL)"));
+  r.push_back(Parameter::Mandatory('o', "output database file"));
+  r.push_back(Parameter::Optional('n', "fully qualified repository name"));
+  r.push_back(Parameter::Optional('k', "repository master key(s) / dir"));
+  r.push_back(Parameter::Optional('l', "temporary directory"));
+  r.push_back(Parameter::Optional('h', "root hash (other than trunk)"));
+  return r;
+}
+
+int CommandFileStats::Main(const ArgumentList &args) {
+  shash::Any manual_root_hash;
+  const std::string &repo_url  = *args.find('r')->second;
+  db_path_ = *args.find('o')->second;
+  const std::string &repo_name =
+    (args.count('n') > 0) ? *args.find('n')->second : "";
+  std::string repo_keys =
+    (args.count('k') > 0) ? *args.find('k')->second : "";
+  if (DirectoryExists(repo_keys))
+    repo_keys = JoinStrings(FindFilesBySuffix(repo_keys, ".pub"), ":");
+  const std::string &tmp_dir   =
+    (args.count('l') > 0) ? *args.find('l')->second : "/tmp";
+  if (args.count('h') > 0) {
+    manual_root_hash = shash::MkFromHexPtr(shash::HexPtr(
+      *args.find('h')->second), shash::kSuffixCatalog);
+  }
+
+  tmp_db_path_ = "/tmp/cvmfs_filestats/";
```

Comment:
Shouldn't this be the user's selected `tmp_dir`?

Reply:
Yes, that makes more sense. I've changed it so that the `cvmfs_filestats` subfolder gets created in the user-defined temporary folder. The folder itself is used by HTTPFetcher as well.